### DISCUSSION TITLE
ISS transit

### DISCUSSION THREAD

**Discussion by delcompan** [cosinekitty/astronomy#230] [General]
The library is great. But I want to calculate the ISS transit with the library, it doesn't seem to have anything about satellite orbits.  I hope you can add this section. Thanks!

**Comment by cosinekitty** [comment]
Hi @delcompan and welcome to the project!  Tracking artificial satellites is really cool. However, it requires constant updates because of unpredictable events:

- variable atmospheric drag and sunlight pressure
- use of maneuvering thrusters to avoid space junk

This is out of scope for Astronomy Engine because its goal is to predict natural objects for centuries in the past and future without external dependencies (no requirement of internet connection, external database, etc). Whoever wants to provide ISS predictions needs to constantly stay on top of tracking its actual position as conditions change, which would be quite a job.

Have you taken a look at [Heavens Above](https://www.heavens-above.com/)? It is a wonderful resource for tracking artificial satellites, including the ISS.

**Comment by delcompan** [comment]
Thanks for your reply.  I have been using skyfield to calculate the transit of the space station to the moon or the sun. I want to make an android app, but skyfield does not support java. Your library has the most language support I've ever seen.

**Comment by cosinekitty** [comment]
That is interesting. I took a look at Skyfield to see how they do this. They do require updating a data set called a TLE.  I read in their [Earth Satellites](https://rhodesmill.org/skyfield/earth-satellites.html) page the following:

> The accuracy of the satellite positions is not perfect. To quote directly from [Revisiting Spacetrack Report #3](https://celestrak.com/publications/AIAA/2006-6753/) Appendix B:
>>“The maximum accuracy for a TLE is limited by the number of decimal places in each field. In general, TLE data is accurate to about a kilometer or so at epoch and it quickly degrades.”

I wonder if the ISS data is somehow better than they state, because being a kilometer off would make transit predictions often wrong. Perhaps the kilometer error is for high altitude satellites, not low earth orbit? Have you tried observing these transits yourself?  I'm curious about any success you have had with it. Very interesting!

**Comment by delcompan** [comment]
I know two websites, “heavens-above.com” for predicting space station or satellite transit observation point, and “transit-finder.com” for predicting transit sun or transit moon. I wrote two python scripts with skyfield to implement the above two functions respectively. The former is in good agreement with heavens-above, while the latter has some functions I can't fully implement yet, but the predicted time of moon or sun transit,  latitude and longitude of the best observation point, target Altitude and Azimuth are basically consistent. And according to the predicted time and latitude and longitude, it can be reproduced in Stellarium.
https://www.bilibili.com/video/BV1ni4y1d7AC?spm_id_from=333.999.0.0
 Here is a video I recorded,the subtitles are in Chinese.

**Reply by RalphHightower** [reply]
NASA has a web page called [Spot the Station](https://spotthestation.nasa.gov/)
https://spotthestation.nasa.gov/

**Comment by delcompan** [comment]
Before running the script each time, the script checks the TLE file to make sure the TLE data are up-to-date. And it is better to set the detection time less than 7 days, otherwise the prediction result is not accurate.

**Reply by adamhill** [reply]
And I bet one could set up a Github Action to pull the TLE file and check timestamps if it is CURL'able :-)

**Comment by cosinekitty** [comment]
Interesting! What is the source of your TLE files?

**Comment by delcompan** [comment]
http://celestrak.com/NORAD/elements/stations.txt
If you just need the TLE data of a particular satellite or space station.
https://celestrak.com/satcat/tle.php?CATNR={ satellite code}
